### ISSUE TITLE
bug: Local Whisper models fail to start on Windows due to binary path mismatch

### ISSUE THREAD

**Issue by shawnriju** [freestyle-voice/freestyle#123]
# Bug Report: Local Whisper Model Fails to Start on Windows

## Description

When attempting to run the application on a Windows system and switching to the local Whisper model, the backend Whisper server fails to start with a message `whisper-server binary not found`. 
When attempting to record anyway, the microphone recording UI elements appear, but audio transcription does not execute.

## Steps to Reproduce

1. Run the Freestyle application on a **Windows** machine.
2. Configure the application to use the **local Whisper model**.
3. Trigger the recording hotkey.
4. Speak into the microphone and release the hotkey.
5. Observe that no transcription text is generated or added to the application history.

## Dev Terminal Logs

Whenever the local model is triggered, the development terminal outputs:

```text
[whisper] Background server start failed: whisper-server binary not found
[whisper] Background server start failed: whisper-server binary not found
[whisper] Downloading pre-built Windows binaries...
[whisper] Windows binaries downloaded
[whisper] transcribe: serverRunning=false serverBinary=false cli=false
```

## Suspected Root Cause

The application downloads the upstream `whisper.cpp` pre-built Windows binaries directly from the official repository:

```text
https://github.com/ggml-org/whisper.cpp/releases/download/v${WIN_VERSION}/whisper-bin-x64.zip
```

By default, the upstream `whisper-bin-x64.zip` archive packages its executables inside a compiled `Release/` subdirectory.

When Freestyle extracts this archive into:

```text
%USERPROFILE%\.cache\freestyle\whisper-bin\
```

the resulting executable path becomes:

```text
...\whisper-bin\Release\whisper-server.exe
```

However, in `apps/server/src/lib/whisper/constants.ts`, the `getBinDir()` function resolves to the root extraction directory:

```ts
export function getBinDir(): string {
  return join(homedir(), ".cache", "freestyle", "whisper-bin");
}
```

As a result, the application looks for `whisper-server.exe` directly inside:

```text
...\whisper-bin\
```

instead of:

```text
...\whisper-bin\Release\
```

This causes the binary lookup to fail, which prevents the local Whisper server from starting.

## Suggested Fix

Update `getBinDir()` in `apps/server/src/lib/whisper/constants.ts` to account for the Windows-specific archive structure:

```ts
export function getBinDir(): string {
  const baseDir = join(homedir(), ".cache", "freestyle", "whisper-bin");
  return process.platform === "win32"
    ? join(baseDir, "Release")
    : baseDir;
}
```

## Verification

Tested locally on a Windows development environment.

After applying the above modification:
`whisper-server` initializes successfully and the Local Whisper transcription works correctly.
